src/p4est_mesh.c

```diff
@@ -1136,6 +1212,36 @@ p4est_mesh_memory_used (p4est_mesh_t * mesh)
   return all_memory;
 }
 
+static void
+mesh_params_init (p4est_mesh_params_t * params)
```

Comment:
I'd be suggesting to work with static parameters. Then we wouldn't need the new and destroy functions, just p4est_mesh_params_init, which first memsets the whole structure to zero and then writes defaults into it.